File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        window = 0
        i = 0

        while i < k:
            window += nums[i]
            i += 1

        start = 0
        end = k
        maxsum = window

        while end < len(nums):
            window -= nums[start]
            start += 1

            window += nums[end]
            end += 1

            maxsum = max(maxsum, window)

        return float(maxsum) / k

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int n= nums.length;
        int window = 0;
        for(int i=0;i<k;i++) window += nums[i];

        int maxsum = window;
        int start = 0;
        for(int i = k; i<n; i++){
            window -= nums[start];
            start++;

            window += nums[i];

            maxsum = Math.max(window, maxsum);
        }
        return (double)maxsum/k;
    }
}